crates/vision/src/luminance_image_extractor.rs

```diff
@@ -0,0 +1,58 @@
+use std::num::NonZeroU32;
+
+use color_eyre::Result;
+use context_attribute::context;
+use fast_image_resize::{DynamicImageView, FilterType, ImageView, ResizeAlg, Resizer};
+use framework::MainOutput;
+use types::{grayscale_image::GrayscaleImage, nao_image::NaoImage};
+
+use crate::CyclerInstance;
+
+pub struct LuminanceImageExtractor {}
+
+#[context]
+pub struct CreationContext {}
+
+#[context]
+pub struct CycleContext {
+    pub instance: CyclerInstance,
+    pub image: Input<NaoImage, "image">,
+}
+
+#[context]
+pub struct MainOutputs {
+    pub luminance_image: MainOutput<GrayscaleImage>,
+}
+
+impl LuminanceImageExtractor {
+    pub fn new(_context: CreationContext) -> Result<Self> {
+        Ok(Self {})
+    }
+
+    pub fn cycle(&mut self, context: CycleContext) -> Result<MainOutputs> {
+        let grayscale_buffer: Vec<_> = context
+            .image
+            .buffer
+            .iter()
+            .flat_map(|pixel| [pixel.y1, pixel.y2])
+            .collect();
+        let y_image = ImageView::from_buffer(
+            NonZeroU32::new(context.image.width()).unwrap(),
+            NonZeroU32::new(context.image.height()).unwrap(),
+            &grayscale_buffer,
+        )?;
+        let dst_width = NonZeroU32::new(80).unwrap();
```

Comment:
Daylight Saving Time?! Dry Sealed Turret?! Delta Sigma Theta?! Department of Science and Technology?! Drive Self Test?! Digital Signal Transponder?! Data Summary Tape?! Decision Support Template?! Debug Symbol Table?! Direct Support Team?! Dynamic String Table?! Defense and Special Teams?! Danish Standard Time?! You get it.

Please don't abbreviate.

Reply:
:sweat: